src/librbd/Journal.cc

```diff
@@ -761,22 +761,19 @@ void Journal<I>::user_flushed() {
 }
 
 template <typename I>
-uint64_t Journal<I>::append_write_event(uint64_t offset, size_t length,
-                                        const bufferlist &bl,
-                                        bool flush_entry) {
-  ceph_assert(m_max_append_size > journal::AioWriteEvent::get_fixed_size());
-  uint64_t max_write_data_size =
-    m_max_append_size - journal::AioWriteEvent::get_fixed_size();
-
+void Journal<I>::create_write_event_entries(Bufferlists &bufferlists,
+                                            const uint64_t max_write_data_size,
+                                            uint64_t offset, size_t length,
+                                            const bufferlist &bl,
+                                            uint64_t buffer_offset) {
```

Comment:
nit: There is a convention to have input params first and output params last, and use `*` for output params. This way it is easier to see what is going to be changed by the func. Also I would name it `append_write_event_entries` or `add_write_event_entries`.

Reply:
OK, renamed to `add_write_event_entries` and hopefully it matches the convention better now!